### DISCUSSION TITLE
Pockels based photonic phase shifters

### DISCUSSION THREAD

**Discussion by lukasc-ubc** [HelgeGehring/femwell#87] [General]
Hi @HelgeGehring 

In my [ELEC463 course](https://ece.ubc.ca/courses/elec-463/), students are designing, fabricating, and testing an electro-optic silicon-organic hybrid polymer modulator.

I am wondering if your tool could be used for such a simulation? 

What we need is:
1) electro-static simulation for the electric field 2D profile, as a function of the applied voltage between two metal conductors. Assume the material is perfectly insulating.
2) extract that 2D distribution, and multiply by the r33 coefficient, to get a ∆n 2D profile
3) mode calculations using Si core and polymer + ∆n cladding
4) sweep voltage, and plot waveguide n_eff

Here is the device cross-section

<img width="514" alt="image" src="https://github.com/HelgeGehring/femwell/assets/15843200/4dd02772-210a-412e-8df2-0a2d6226686e">

Thank you very much,
Lukas

**Comment by joamatab** [comment]
Hi Lukas,

you can take a look at this example
https://helgegehring.github.io/femwell/photonics/examples/culomb.html

**Reply by HelgeGehring** [reply]
@joamatab thanks for answering so quickly!

Yes, that example is super close to the electro-optic silicon-organic hybrid polymer modulator!

Probably the small difference is perfect for the students to transfer from the example to the polymer modulator?

**Comment by lukasc-ubc** [comment]
Thank you very much!  I think that is a great starting point.

**Comment by lukasc-ubc** [comment]
Hi @HelgeGehring 

Looking through this example, I see you define epsilon = basis_epsilon_r.zeros() + 1.445.  Shouldn't there be a **2 to convert from index of refraction to permittivity?  I do see the effective index is being correctly, so perhaps it is just a nomenclature problem?  Any way this could be corrected?

And you have different values of epsilon -- at the top these are for DC electric fields, versus lower down for optical?  

Thank you
Lukas

**Reply by HelgeGehring** [reply]
@lukasc-ubc I've updated the example (see https://github.com/HelgeGehring/femwell/blob/main/docs/photonics/examples/culomb.py#L96, will be online in the docs in 1h after CICD ran through).

The epsilon was squared later on. Now the variable is called refractive_index and is squared when it's given as a parameter to calcualte_modes. Like this everything should be correct from the nomenclature, right?

And yes, the epsilons at the top are for the dc electric fields! Maybe we should name them differently to clarify things?
`epsilon_electrical` and `refractive_index_optical`? Or do you have a better suggestion?

**Comment by lukasc-ubc** [comment]
I think your change helps clarify, though the mixture of epsilon and index are a bit confusing.
Somehow it would be nice for the material properties to have separate variables, and the solvers to use either the DC electric field versus the THz optical one.
But maybe for a solver that can be used for many applications, it is ok to let the user be responsible for knowing which one to use.

thank you

**Reply by HelgeGehring** [reply]
Thanks for the feedback! I updated the variable names to further clarify things!
Let me know if further changes would help! Happy to improve it further :)
